marimo/_ast/cell.py

```diff
@@ -148,111 +144,211 @@ def set_status(self, status: CellStatusType) -> None:
-        glbls = {}
-        glbls.update(defaults)
-        pos = 0
-        for arg in args:
-            glbls[parameters[pos]] = arg
-            pos += 1
-        if pos < n_args:
-            raise TypeError(
-                f.__name__
-                + f"() missing {n_args - pos} required arguments: "
-                + " and ".join(f"'{p}'" for p in parameters[pos:n_args])
-            )
-
-        for kwarg, value in kwargs.items():
-            if kwarg not in parameters:
-                raise TypeError(
-                    f.__name__
-                    + "() got an unexpected keyword argument '{kwarg}'"
-                )
-            else:
-                glbls[kwarg] = value
-        return glbls
-
-    def _returns(glbls: dict[Any, Any]) -> tuple[Any, ...]:
-        return tuple(glbls[name] for name in return_names)
-
-    # Wrapper for executing cell using the function's signature.
-    #
-    # Alternative for passing a globals dict
-    #
-    # we use execute_cell instead of calling `f` directly because
-    # we want to obtain the cell's HTML output, which is the last
-    # expression in the cell body.
-    #
-    # TODO: stash output if mo.collect_outputs() context manager is active
-    #       ... or just make cell execution return the output in addition
-    #       to the defs, which might be weird because that doesn't
-    #       match the function signature
-    if inspect.iscoroutinefunction(f):
-
-        @functools.wraps(f)
-        async def func(*args: Any, **kwargs: Any) -> tuple[Any, ...]:
-            glbls = _prepare_args(*args, **kwargs)
-            _ = await execute_cell_async(cell, glbls)
-            return _returns(glbls)
+            output, defs = ret
+        ```
 
-    else:
 
-        @functools.wraps(f)
-        def func(*args: Any, **kwargs: Any) -> tuple[Any, ...]:
-            glbls = _prepare_args(*args, **kwargs)
-            _ = execute_cell(cell, glbls)
-            return _returns(glbls)
+        **Arguments**:
+
+        - You may pass values for any of this cell's references as keyword
+          arguments. marimo will automatically compute values for any refs
+          that are not provided by executing the parent cells that compute them
 
-    cell_func = cast(CellFunction[CellFuncTypeBound], func)
-    cell_func.cell = cell
-    cell_func.args = args
-    cell_func.code = code
-    return cell_func
+        **Returns**:
+
+        - a tuple `(output, defs)`, or an awaitable of the same, where `output`
+          is the cell's last expression and `defs` is a `dict` mapping the
+          cell's defined names to their values.
+        """
+        assert self._app is not None
+        if self._is_coroutine():
+            return self._app.run_cell_async(cell=self, kwargs=refs)
+        else:
+            return self._app.run_cell_sync(cell=self, kwargs=refs)
+
+    def __call__(self, *args: Any, **kwargs: Any) -> None:
```

Comment:
Decided against implementing this to avoid confusion, given that the signature of `run` doesn't match the signature of the cell in the file format.